src/app/(proper_react)/(onboarding)/user/welcome/EnterInfo.tsx

```diff
@@ -4,22 +4,357 @@
+    { type: "dialog" },
+    explainerDialogState
+  );
+
+  const confirmDialogState = useOverlayTriggerState({});
+  const confirmDialogTrigger = useOverlayTrigger(
+    { type: "dialog" },
+    confirmDialogState
+  );
+
+  const l10n = useL10n();
+
+  const userDetailsData = [
+    {
+      label: l10n.getString("onboarding-enter-details-label-first-name"),
+      key: "firstName",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-first-name"
+      ),
+      value: firstName,
+      displayValue: firstName,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setFirstName,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-last-name"),
+      key: "lastName",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-last-name"
+      ),
+      value: lastName,
+      displayValue: lastName,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setLastName,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-location"),
+      key: "location",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-location"
+      ),
+      value: location,
+      displayValue: location,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setLocation,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-date-of-birth"),
+      key: "dateOfBirth",
+      type: "date",
+      placeholder: "",
+      value: dateOfBirth,
+      displayValue: new Date(dateOfBirth).toLocaleDateString("en-US"),
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setDateOfBirth,
+    },
+  ];
+
+  const validateEnteredInfo = () => {
+    const invalidInputKeys = userDetailsData
+      .filter(({ value }) => !getIsValidInfo(value))
+      .map(({ key }) => key);
+
+    setInvalidInputs(invalidInputKeys);
+  };
+
+  useEffect(() => {
+    if (!hasMounted.current) {
+      hasMounted.current = true;
+      return;
+    }
+
+    if (!invalidInputs?.length) {
+      confirmDialogState.open();
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [invalidInputs]);
+
+  // TODO: Move dialogs into its own component
+  const WhyDoWeNeedInfoDialog = () => (
+    <Dialog illustration={<Image src={whyWeNeedInfoHero} alt="" />}>
+      <div
+        className={`${viewStyles.dialogContents} ${enterInfoStyles.dialogContents}`}
+      >
+        <p>{l10n.getString("onboarding-enter-details-dialog")}</p>
+        <strong>
+          {l10n.getString("onboarding-enter-details-diaglog-subheader")}
+        </strong>
+
+        <ol className={viewStyles.list}>
```

Comment:
Contrary to what the designs currently show I proposed to use an ordered list here similar to the dialog in the “Get Started” step. Question is pending with UX: https://www.figma.com/file/CaEKIhvSJqf6KNIMzSkt40/Concepts-for-Monitor-MVP-Redesign?type=design&node-id=1398-74631&mode=design&t=wvp4wSfzJDytJyvz-0.